### PR TITLE
Exclude sign signature if it is empty - Closes #2973

### LINKED ISSUE #2973: API Transactions exclude field signSignature if its empty
### Expected behavior
`signSignature` field should be removed if it's empty and only be shown when it has value.

### Actual behavior
`signSignature` field shows empty string if not present, this can lead to confusion, so we should not show `signSignature` if its empty.
### Steps to reproduce
`1.5.0-rc.1`: http://192.154.224.145:7000/api/transactions/?sort=timestamp:desc&type=2&limit=100&offset=0&fromTimestamp=0
`< 1.5.0-rc.1`: http://lindsay.lisk.plus:8000/api/transactions/?sort=timestamp:desc&type=2&limit=100&offset=0&fromTimestamp=0
### Which version(s) does this affect? (Environment, OS, etc...)
1.5.0

### PR DESCRIPTION
**Pull request by diego-G**
### Review checklist

* The PR resolves #2973
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: api/controllers/transactions.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ function transactionFormatter(transaction) {
 	result.senderId = result.senderId || '';
 	result.recipientId = result.recipientId || '';
 	result.recipientPublicKey = result.recipientPublicKey || '';
-	result.signSignature = result.signSignature || '';
+	result.signSignature = result.signSignature || undefined;
```
Comment (MaciejBaj) [inline api/controllers/transactions.js]:
```suggestion
	if (!result.signSignature) {
		delete result.signSignature;
	}
```
Np?

Comment (diego-G) [reply api/controllers/transactions.js]: The suggestion has the same effect than the solution. Swagger takes care of not showing it when it is undefined.

**Comment by diego-G** [comment]
yes @ManuGowda, we recovered the former behaviour.

Regarding the tests, Swagger schema validator does the job.

**Comment by ManuGowda** [comment]
how does swagger schema validates of the `signSignature` is present or not? so swagger will ensure the behavior? I do not want to keep this as blocker, I know we want hurry and rush for PR to be merged, if we identify and there is a fix for the bug without the coverage of test I am not sure about it, @MaciejBaj whats your call? not for this PR for any future PR.

**Comment by MaciejBaj** [comment]
If Swagger would do the job, we wouldn't see this problem. 

@diego-G please extend the second signature tests to validate if the `signSignature` field is there only when the second signature is present.

**Comment by MaciejBaj** [comment]
@ManuGowda covering the bug with proper test scenario is the behaviour we definitely want to keep, there is no excuse in being in rush in this case.

**Comment by ManuGowda** [comment]
@michielmulders are you taking care of the tests here?

**Comment by michielmulders** [comment]
Yes I am!

File: test/functional/http/get/transactions.js [hunk 2]
```diff
@@ -1040,6 +1049,54 @@ describe('GET /api/transactions', () => {
...
+
+				await sendTransactionPromise(transaction, 200);
+				await waitFor.confirmations([transaction.id]);
+
+				// Act
+				const {
+					body: { data: transactions },
+				} = await transactionsEndpoint.makeRequest({ id: transaction.id }, 200);
+				// Assert
+				expect(transactions[0].signSignature).to.not.be.empty;
```
Comment (ManuGowda) [inline test/functional/http/get/transactions.js]: this looks good, what would be nice to have is comparing `signSignature` against the expecting `signSignature`
something like `expect(transactions[0].signSignature).to.eq(signSignatureFuc(required params))`. but for now i am goona approve, you can use this pattern for future, this way the expectation is spot on.

Comment (michielmulders) [reply test/functional/http/get/transactions.js]: Ok, thanks!

**Pull request merged**